src/decision-engine/req-queue.js

```diff
@@ -254,21 +274,31 @@ class PeerTasks {
 
   /**
    * Remove pending tasks with the given topic
+   *
    * @param {string} topic
+   * @returns {void}
    */
   remove (topic) {
     this._pending.delete(topic)
   }
 
   /**
    * No work to be done, this PeerTasks object can be freed.
+   *
    * @returns {boolean}
    */
   isIdle () {
-    return this._pending.length === 0 && this._active.length === 0
+    return this._pending.length === 0 && this._active.size === 0
```

Comment:
It looks like there was a bug in the code, because `_active` is a `Set` instance which doesn't have `length` property.